src/setup_node_env/harden/index.js

```diff
@@ -6,5 +6,35 @@
  * Side Public License, v 1.
  */
 
-require('./child_process');
-require('./lodash_template');
+var ritm = require('require-in-the-middle');
+var lodashPatch = require('./lodash_template');
+var patchChildProcess = require('./child_process');
+
+// the performance cost of using require-in-the-middle is atm directly related to the number of
+// registered hooks (as require is patched once for EACH hook)
+// This is why we are defining a single hook delegating for each of the patches we need to apply
+new ritm.Hook(
+  ['child_process', 'lodash', 'lodash/template', 'lodash/fp', 'lodash/fp/template'],
+  function (module, name) {
+    switch (name) {
+      case 'child_process': {
+        return patchChildProcess(module);
+      }
+      case 'lodash': {
+        module.template = lodashPatch.createProxy(module.template);
+        return module;
+      }
+      case 'lodash/template': {
+        return lodashPatch.createProxy(module);
+      }
+      case 'lodash/fp': {
+        module.template = lodashPatch.createFpProxy(module.template);
+        return module;
+      }
+      case 'lodash/fp/template': {
+        lodashPatch.createFpProxy(module);
```

Comment:
Random drive-by (i.e. feel free to totally ignore me, I'm here from RITM-land): Is the fall-through in the case intentional? It works, but could be a surprise if more cases are added later.

Reply:
woups, good catch, no it wasn't intentional, I will fix it thanks